### PR TITLE
checkpoint system implementation

### PR DESCRIPTION
**Pull request by hawciu**
very basic checkpoint system with a placeholder map for further track development

### CODE REVIEW

File: Big-Bumper-Bash/Assets/CheckpointSystem/CheckpointManagerScript.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        else
+        {
+            currentCheckpoint++;
+            checkpoints[currentCheckpoint].SetActive(true);
+        }
+    }
+
+    void RaceFinished()
+    {
+    }
```
Comment (Gathaspar66) [inline Big-Bumper-Bash/Assets/CheckpointSystem/CheckpointManagerScript.cs:47]: Add appropriate commentary on what this method will do in the future.

File: Big-Bumper-Bash/Assets/CheckpointSystem/CheckpointManagerScript.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    {
+        CheckpointManager = this;
+        if (checkpoints.Count > 0)
+        {
+            foreach (GameObject checkpoint in checkpoints)
+            {
+                checkpoint.SetActive(false);
+            }
+            checkpoints[0].SetActive(true);
+        }
```
Comment (Gathaspar66) [inline Big-Bumper-Bash/Assets/CheckpointSystem/CheckpointManagerScript.cs:23]: Isolate this to a separate method

**Review by Gathaspar66 — COMMENTED** [review]
Generally good.

File: Big-Bumper-Bash/Assets/CheckpointSystem/CheckpointManagerScript.cs [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    void Update()
+    {
+        
+    }
+
+    public void CheckpointReached()
+    {
+        if (currentCheckpoint == checkpoints.Count - 1)
+        {
+            RaceFinished();
```
Comment (klaudix30) [inline Big-Bumper-Bash/Assets/CheckpointSystem/CheckpointManagerScript.cs:36]: methods should be in verb form

File: Big-Bumper-Bash/Assets/CheckpointSystem/CheckpointManagerScript.cs [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        }
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        
+    }
+
+    public void CheckpointReached()
```
Comment (masslany) [inline Big-Bumper-Bash/Assets/CheckpointSystem/CheckpointManagerScript.cs:32]:
```suggestion
    public void OnCheckpointReached()
```

File: Big-Bumper-Bash/Assets/CheckpointSystem/CheckpointManagerScript.cs [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    {
+        if (currentCheckpoint == checkpoints.Count - 1)
```
Comment (masslany) [inline Big-Bumper-Bash/Assets/CheckpointSystem/CheckpointManagerScript.cs:34]:
```suggestion
        if (isLastCheckpointReached())
```
and create a private method to encapsulate that logic.

File: Big-Bumper-Bash/Assets/CheckpointSystem/CheckpointManagerScript.cs [hunk 6]
```diff
@@ -0,0 +1,48 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class CheckpointManagerScript : MonoBehaviour
+{
+    public static CheckpointManagerScript CheckpointManager;
```
Comment (masslany) [inline Big-Bumper-Bash/Assets/CheckpointSystem/CheckpointManagerScript.cs:7]:
```suggestion
    public static CheckpointManagerScript checkpointManager;
```

File: Big-Bumper-Bash/Assets/CheckpointSystem/CheckpointManagerScript.cs [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+    public List<GameObject> checkpoints= new();
```
Comment (masslany) [inline Big-Bumper-Bash/Assets/CheckpointSystem/CheckpointManagerScript.cs:9]:
```suggestion
    public List<GameObject> checkpoints = new();
```
nit: formatting

File: Big-Bumper-Bash/Assets/CheckpointSystem/CheckpointManagerScript.cs [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+        }
+    }
+
+    bool IsLastCheckpointReached()
+    {
+        if (currentCheckpoint == checkpoints.Count - 1)
+        {
+            return true;
+        }
+        return false;
```
Comment (masslany) [inline Big-Bumper-Bash/Assets/CheckpointSystem/CheckpointManagerScript.cs:48]:
```suggestion
        return currentCheckpoint == checkpoints.Count - 1;
```

**Review by Gathaspar66 — APPROVED** [review]
lgtm!

**Review by masslany — APPROVED** [review]

**Pull request merged**